src/libraries/System.Drawing.Common/src/System/Drawing/ImageAnimator.cs

```diff
@@ -381,10 +382,20 @@ public static void StopAnimate(Image image, EventHandler onFrameChangedHandler)
         private static void AnimateImages50ms()
         {
             Debug.Assert(s_imageInfoList != null, "Null images list");
+            DateTime animationStart = DateTime.Now;
+            double totalAnimationTime, totalAnimationTimeLastAnimation = 0;
+            int thisAnimationDuration;
 
             while (true)
             {
-                // Acquire reader-lock to access imageInfoList, elemens in the list can be modified w/o needing a writer-lock.
+                Thread.Sleep(50);
```

Comment:
>Sleep [](http://example.com/codeflow?start=23&length=5)

I guess this can cause animation be not smooth especially when having some busy machine. no?

Reply:
interesting I am seeing the original code is using it. So, I am not sure if the current behavior is good enough.

Reply:
Possibly, yes. Animation delays are in increments of 1/100ths of a second, If we want smoother animation, we could reduce the sleep duration. With the severity the animation drift that existed and the lack of customer reports of the issues though, I chose to leave the sleep duration as it was. So the smoothness should remain basically as-is, but now with accurate timing.

Reply:
FWIW, nearly every GIF playback software I've encountered limits playback speed to 4/100ths of a second per frame.  Anything less is generally too expensive, and that's fast enough to show film/video clips at roughly correct speed (as close as you can get in even hundredths of a second anyway).

Reply:
Thanks, @saucecontrol. Do you think it's valuable to reduce the sleep to 40ms to align with that?

Reply:
Depends on your appetite for change, but yeah, I think it would improve user experience for common GIF usage.